### PR TITLE
make the testjob source dir configurable

### PR DESCRIPTION
**Pull request by bmeng**
### What type of PR is this?

_feature_

### What this PR does / Why we need it?

Currently, we have the ocm backplane testjob command to test the script with custom image, but it has to be run under the script dir
Add the new parameter to specify the source dir for the script

### Which Jira/Github issue(s) does this PR fix?

_Resolves #_

### Pre-checks (if applicable)

- [x] Ran unit tests locally
- [x] Validated the changes in a cluster
- [ ] Included documentation changes with PR

### CODE REVIEW

File: cmd/ocm-backplane/testJob/createTestJob.go [hunk 1]
```diff
@@ -208,11 +226,14 @@ func runCreateTestJob(cmd *cobra.Command, args []string) error {
...
+func createTestScriptFromFiles(workingDir string, dryRun bool) (*backplaneApi.CreateTestScriptRunJSONRequestBody, error) {
+
+	metaFile := workingDir + "metadata.yaml"
+
 	// Read the yaml file from cwd
-	yamlFile, err := os.ReadFile("metadata.yaml")
+	yamlFile, err := os.ReadFile(metaFile)
 	if err != nil {
-		logger.Errorf("Error reading metadata yaml: %v, ensure you are in a script directory", err)
+		logger.Errorf("Error reading metadata yaml: %v, ensure either you are in a script directory or specified the corrent source dir", err)
```
Comment (MitaliBhalla) [inline cmd/ocm-backplane/testJob/createTestJob.go:236]:
```suggestion
		logger.Errorf("Error reading metadata yaml: %v, ensure either you are in a script directory or you have specified the correct source dir", err)
```

**Review by MitaliBhalla — CHANGES_REQUESTED** [review]

**Review by hectorakemp — COMMENTED** [review]
Great work @bmeng !
I think only one change is needed for an `lgtm` from me. We need a test for the negative case of when the specified source dir doesn't exist on the file system. Then it's `lgtm`!

**Comment by bmeng** [comment]
@hectorakemp thanks for review.
Have added a new function `checkDirectory()` to ensure the given directory is valid. And related test added.

**Review by hectorakemp — APPROVED** [review]
/lgtm
/approve
